libkineto/src/CuptiActivityProfiler.cpp

```diff
@@ -191,6 +191,17 @@ std::ostream& operator<<(std::ostream& oss, const CuptiActivityProfiler::ErrorCo
   return oss;
 }
 
+CuptiActivityProfiler::~CuptiActivityProfiler() {
+  if(stopByIterThread) {
+    std::lock_guard<std::mutex> guard(mutex_);
```

Comment:
why is locking needed here ? to avoid double free?

Reply:
Yes, I checked the destruction sequence,  there's no race condition here. `ActivityProfilerController` will wait thread `profilerThread_` to quit in its destruction method, then, there's no thread accessing `stopByIterThread`.